xbmc/epg/EpgContainer.cpp

```diff
@@ -235,20 +235,16 @@ void CEpgContainer::LoadFromDB(void)
 
 bool CEpgContainer::PersistTables(void)
 {
-  m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
-  m_critSection.unlock();
-  return m_database.Persist(copy);
+  CSingleLock lock(m_critSection);
+  return m_database.Persist(m_epgs);
 }
 
 bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
-  m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
-  m_critSection.unlock();
+  CSingleLock lock(m_critSection);
```

Comment:
same as above, no holding locks here while persisting. try this on a pi with a slow sd card and lots of entries of you need a reason